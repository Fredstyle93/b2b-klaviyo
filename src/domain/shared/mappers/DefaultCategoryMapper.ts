import { Category } from '@commercetools/platform-sdk';
import { CategoryMapper } from './CategoryMapper';

export class DefaultCategoryMapper implements CategoryMapper {
    public mapCtCategoryToKlaviyoCategory(category: Category): CategoryRequest {
        return {
            data: {
                type: 'catalog-category',
                attributes: {
                    external_id: category.id,
                    name: category.name[Object.keys(category.name)[0]],
                    integration_type: '$custom',
                    catalog_type: '$default',
                },
            },
        };
    }
}
