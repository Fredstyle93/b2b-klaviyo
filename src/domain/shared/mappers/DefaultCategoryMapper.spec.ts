import { DefaultCategoryMapper } from "./DefaultCategoryMapper";
import { sampleCategoryCreatedMessage } from "../../../test/testData/ctCategoryMessages";

const categoryMapper = new DefaultCategoryMapper();

describe("map CT category to Klaviyo category", () => {
  it("should map a commercetools category to a klaviyo category", () => {
    const klaviyoEvent = categoryMapper.mapCtCategoryToKlaviyoCategory(sampleCategoryCreatedMessage.category);
    expect(klaviyoEvent).toMatchSnapshot();
  })
})