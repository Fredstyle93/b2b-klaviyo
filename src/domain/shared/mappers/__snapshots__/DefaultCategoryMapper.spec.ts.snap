// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map CT category to Klaviyo category should map a commercetools category to a klaviyo category 1`] = `
{
  "data": {
    "attributes": {
      "catalog_type": "$default",
      "external_id": "cd2cd62b-7f08-44e4-96a8-d3e34c40cac9",
      "integration_type": "$custom",
      "name": "MessageTriggerTest1",
    },
    "type": "catalog-category",
  },
}
`;
